problems/2017-03-30-side-effects-may-include/index.js

```diff
@@ -0,0 +1,114 @@
+/*
+  Being the excellent HR person with a decent amount of JavaScript knowledge, 
+  you decided that you'll keep track of employees using an array in JS.
+
+  You've put all the employees in the array, based on the order in which they were hired.
+
+  Note: This is actually a really bad way to keep track of people in general, but you already knew that.
+*/
+const employees = [
+  {
+    name : 'Daniel',
+    dateOfBirth : '1984-01-12'
+  },
+  {
+    name : 'Jack',
+    dateOfBirth : '1946-04-06'
+  },
+  {
+    name : 'Jane',
+    dateOfBirth : '1989-08-30'
+  },
+  {
+    name : 'Bill',
+    dateOfBirth : '1965-11-06'
+  },
+  {
+    name : 'Samantha',
+    dateOfBirth : '1995-06-24'
+  },
+  {
+    name : 'Bae',
+    dateOfBirth : '1978-03-30'
+  }
+];
+
+/*
+  This function is pure. No external state is used, 
+  and results are always consistent.
+*/
+const getYearDuration = function(timeInMillis) {
+  return Math.floor(timeInMillis / 1000 / 60 / 60 / 24 / 365);
+};
+
+/*
+  This function isn't so pure. 
```

Comment:
Assuming the functions that are externalized are things we don't want them to have to implement, that makes sense to me.